### PR TITLE
:truck: Move defines to there proper files

### PR DESCRIPTION
**Pull request by AJenbo**

### CODE REVIEW

File: Source/diablo.cpp [hunk 1]
```diff
@@ -14,6 +14,13 @@
 
 namespace devilution {
 
+#ifndef DEFAULT_WIDTH
+#define DEFAULT_WIDTH 640
+#endif
+#ifndef DEFAULT_HEIGHT
+#define DEFAULT_HEIGHT 480
+#endif
```
Comment (julealgon) [inline Source/diablo.cpp:22]: Shouldn't these be in `diablo.h` instead?

Also, why the `#ifndef` around them? Can we remove?

File: Source/encrypt.cpp [hunk 2]
```diff
@@ -16,7 +16,7 @@ void Decrypt(DWORD *castBlock, DWORD size, DWORD key)
 
 	seed = 0xEEEEEEEE;
 	for (i = 0; i < (size >> 2); i++) {
-		DWORD t = SwapLE32(*castBlock);
+		DWORD t = SDL_SwapLE32(*castBlock);
```
Comment (julealgon) [inline Source/encrypt.cpp:19]: This looks unrelated to moving macros?

File: Source/items.h [hunk 3]
```diff
@@ -11,6 +11,12 @@
 
 namespace devilution {
 
+#define MAXRESIST 75
```
Comment (julealgon) [inline Source/items.h:14]: This should be moved to `player.h`.

File: Source/loadsave.cpp [hunk 4]
```diff
@@ -1746,6 +1746,10 @@ void SaveHeroItems(PlayerStruct *pPlayer)
 	SaveItems(&file, pPlayer->SpdList, MAXBELTITEMS);
 }
 
+// 256 kilobytes + 3 bytes (demo leftover) for file magic (262147)
+// final game uses 4-byte magic instead of 3
+#define FILEBUFF ((256 * 1024) + 3)
```
Comment (julealgon) [inline Source/loadsave.cpp:1751]: Isn't it common to have the defines closer to the top of the file?

File: Source/monster.cpp [hunk 5]
```diff
@@ -11,6 +11,12 @@
 
 namespace devilution {
 
+#define NIGHTMARE_TO_HIT_BONUS 85
+#define HELL_TO_HIT_BONUS 120
+
+#define NIGHTMARE_AC_BONUS 50
+#define HELL_AC_BONUS 80
```
Comment (julealgon) [inline Source/monster.cpp:18]: Same as the other one, I always thought constants and defines were to be placed in the header, not the cpp. Was I incorrect?

File: Source/objects.h [hunk 6]
```diff
@@ -9,6 +9,8 @@
 
 namespace devilution {
 
+#define MAX_LVLS 24
```
Comment (julealgon) [inline Source/objects.h:12]: Isn't this level related? Why objects.h?

File: Source/quests.cpp [hunk 7]
```diff
@@ -652,6 +653,8 @@ void SetReturnLvlPos()
 		ReturnLvl = quests[Q_SCHAMB]._qlevel;
 		ReturnLvlT = DTYPE_CATACOMBS;
 		break;
+	case SL_MAZE:
+		break;
```
Comment (julealgon) [inline Source/quests.cpp:657]: Unrelated

File: defs.h [hunk 8]
```diff
@@ -4,30 +4,9 @@
...
-#define GMENU_ENABLED			0x80000000
-
-// must be unsigned to generate unsigned comparisons with pnum
-#define MAX_PLRS				4
-
-#define MAX_CHARACTERS			99
-#define MAX_LVLS				24
-#define MAX_LVLMTYPES			24
 #define MAX_SPELLS				52
 #define MAX_SPELL_LEVEL			15
```
Comment (julealgon) [inline defs.h:8]: Why didn't we move some of these?

**Review by julealgon — CHANGES_REQUESTED** [review]

File: Source/diablo.cpp (hunk 1 above)
Comment (AJenbo) [reply Source/diablo.cpp:22]: its only used in diablo.pp, some devices overwrite it because of there screen size and not having hardware capable of upscaling.

File: Source/encrypt.cpp (hunk 2 above)
Comment (AJenbo) [reply Source/encrypt.cpp:19]: it removed an old dev macro that doesn't make sens in devx.

File: Source/loadsave.cpp (hunk 4 above)
Comment (AJenbo) [reply Source/loadsave.cpp:1751]: if its only used in a specific function i think it makes more sens to have it next to it

File: Source/monster.cpp (hunk 5 above)
Comment (AJenbo) [reply Source/monster.cpp:18]: same

File: Source/objects.h (hunk 6 above)
Comment (AJenbo) [reply Source/objects.h:12]: it appears to only be used to define placment of shrines.

File: Source/quests.cpp (hunk 7 above)
Comment (AJenbo) [reply Source/quests.cpp:657]: yeah, just got rid of a warning to make it cleare to see if i caused any issues

File: defs.h (hunk 8 above)
Comment (AJenbo) [reply defs.h:8]: more complicated to move, save them for round 2

File: Source/dead.h [hunk 9]
```diff
@@ -5,8 +5,12 @@
  */
 #pragma once
 
+#include <SDL.h>
```
Comment (glebm) [inline Source/dead.h:8]: Is this just for the `Uint8`? Would probably be better to switch that `Uint8` to `std::uint8_t` instead. Anyway, it's a minor thing.

**Review by glebm — APPROVED** [review]

File: Source/dead.h (hunk 9 above)
Comment (AJenbo) [reply Source/dead.h:8]: Yeah, the good news is that this makes it clear what files this should be cleaned up in :) I decided to do so at a later point.

File: Source/objects.h (hunk 6 above)
Comment (julealgon) [reply Source/objects.h]: Shouldn't it still be moved to a level-related header though? I think what matters most is the semantic of the value, not its use.

Comment (AJenbo) [reply Source/objects.h]: We could rename it SHRINE_END_LEVEL if that would help.
